### ISSUE TITLE
Jenkins pipeline setup does not copy the type of the predefined validation stamps

### ISSUE THREAD

**Issue by dcoraboeuf** [nemerosa/ontrack#1082]
Given a [Jenkins pipeline](https://github.com/nemerosa/ontrack-jenkins-cli-pipeline) using the [auto promotions](https://github.com/nemerosa/ontrack-jenkins-cli-pipeline/blob/main/vars/ontrackCliSetup.md#promotions)
and given the promotion uses a validation stamp not being defined in the same pipeline

```groovy
ontrackCliSetup(
        promotions: [
                BRONZE: [
                        validations: [
                                "build"
                        ]
                ],
        ]
)
```

and given the validation is predefined with a given type ("test summary" for example)
then I expect the validation to be created into the branch with the same type 
however, the validation is created, but without a type

**Comment by dcoraboeuf** [comment]
Whenever a validation is declared only at the promotion level, the actual call looks like:

```graphql
setupValidationStamp(input: {
    project: $project,
    branch: $branch,
    validation: $validation,
    description: $description,
    dataType: $dataType,
    dataTypeConfig: $dataTypeConfig
})
```

where `dataType` and `dataTypeConfig` are set to `null`.

This works fine for a new validation stamp (the predefined data will be used) but in the case I've seen here, the validation stamp has already been defined at branch level (with a type), and the code happens then to not rely on the predefined validation stamp.

So in the end, it works as defined.

**Comment by dcoraboeuf** [comment]
The workaround is to declare the type explicitly, for example:

```groovy
ontrackCliSetup(
        validations: [
            [
                name: "build",
                tests: [ warningIfSkipped: false]
            ]
        ]
        promotions: [
                BRONZE: [
                        validations: [
                                "build"
                        ]
                ],
        ]
)
```

**Comment by dcoraboeuf** [comment]
Available in 4.5.6
